### ISSUE TITLE
Ada support?

### ISSUE THREAD

**Issue by 1ma** [richfelker/musl-cross-make#133]
The `config.mak.dist` states that the only languages supported are C and C++, and indeed Ada is broken as-is.

However I've been able to build GCC 9.4.0 with Ada support applying a subset of patches from the [Alpine Linux gcc port](https://gitlab.alpinelinux.org/alpine/aports/-/tree/master/main/gcc). I don't know how stable it is, but I've been able to build several important Ada projects with it (XmlAda, GPRBuild, GNATColl and Alire).

Are you interested in these patches?

**Comment by richfelker** [comment]
Yes, I am. It's hard to bootstrap (you need a host compiler that supports Ada to begin with) which means you can't use an existing lineage of MCM toolchains to build it, but it would still be worthwhile to have.

**Comment by kwinsch** [comment]
Ada works on musl with GCC 16.2.0 in our fork (x86_64, aarch64, riscv64): https://github.com/kwinsch/musl-cross-make, patches `0016`–`0019` under `patches/gcc-16.2.0` (three Alpine ports plus a weak import for `pthread_mutexattr_setprioceiling`, which breaks every static tasking link). On bootstrapping: only the first x86_64 stage needs a host GNAT; stage 2 builds Ada for any target from that cross GNAT alone. Prebuilt toolchains are under Releases.
